Add confirm modal on module update

NeOMakinG opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Updating module without confirm leads to bugs and problems when updating a module in production
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #14039.
| How to test?  | Go on module page, click on "Update" on a module out of date, also try the "Update All" on the Update tab

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/22316)
<!-- Reviewable:end -->

Commit:
Add confirm modal before upgrade of a module

Commit:
Add maintenanceMode conditions for message and new button, possibility to add custom buttons to a modal component

Commit:
Change confirm modal id

Commit:
Add translations and maintenance from backend

Commit:
Lint js

Commit:
Get configuration from PS_SHOP_ENABLE

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Module/common.html.twig:
```diff
@@ -32,7 +32,21 @@
     };
+
+    var moduleTranslations = {
+      'singleModuleModalUpdateTitle': '{{ "Are you sure you want to upgrade this module?"|trans({}, 'Admin.Modules.Feature') }}',
```
```suggestion
      'singleModuleModalUpdateTitle': '{{ "Are you sure you want to upgrade this module?"|trans({}, 'Admin.Modules.Notification') }}',
```

Julievrz replied:
```suggestion
      'singleModuleModalUpdateTitle': '{{ "Are you sure you want to upgrade this module?"|trans({}, 'Admin.Modules.Notification') }}',
```

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Module/common.html.twig:
```diff
@@ -32,7 +32,21 @@
+
+    var moduleTranslations = {
+      'singleModuleModalUpdateTitle': '{{ "Are you sure you want to upgrade this module?"|trans({}, 'Admin.Modules.Feature') }}',
+      'multipleModuleModalUpdateTitle': '{{ "Are you sure you want to upgrade these modules?"|trans({}, 'Admin.Modules.Feature') }}',
```
```suggestion
      'multipleModuleModalUpdateTitle': '{{ "Are you sure you want to upgrade these modules?"|trans({}, 'Admin.Modules.Notification') }}',
```

Julievrz reviewed (commented):
Hi, just a little modification of translation domains. Thanks!

Commit:
Change domain translation

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

Commit:
Change domain translation

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

Change domain of translation

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

Revert "Merge branch 'issue14039' of github.com:NeOMakinG/PrestaShop into issue14039"

This reverts commit 562de7ce49ba75076e4f24aea89260930073f332, reversing
changes made to 4c38afa6c443c9ea99a7e0672a8a2d29a0f961ac.

Update domains of translation

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Module/common.html.twig:
```diff
@@ -32,7 +32,21 @@
+      'moduleModalUpdateMaintenance': '{{ "Go to maintenance page"|trans({}, 'Admin.Actions') }}',
+      'moduleModalUpdateUpgrade': '{{ "Upgrade"|trans({}, 'Admin.Actions') }}',
+      'multipleModuleModalUpdateUpgrade': '{{ "Upgrade anyway"|trans({}, 'Admin.Actions') }}',
+      'moduleModalUpdateConfirmMessage': '{{ "We strongly advise you to upgrade the modules on maintenance mode to avoid any cache issues."|trans({}, 'Admin.Modules.Feature') }}',
```
```suggestion
      'moduleModalUpdateConfirmMessage': '{{ "We strongly advise you to upgrade the modules on maintenance mode to avoid any cache issues."|trans({}, 'Admin.Modules.Notification') }}',
```

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Module/common.html.twig:
```diff
@@ -32,7 +32,21 @@
+      'moduleModalUpdateCancel': '{{ "Cancel"|trans({}, 'Admin.Actions') }}',
+      'moduleModalUpdateMaintenance': '{{ "Go to maintenance page"|trans({}, 'Admin.Actions') }}',
+      'moduleModalUpdateUpgrade': '{{ "Upgrade"|trans({}, 'Admin.Actions') }}',
+      'multipleModuleModalUpdateUpgrade': '{{ "Upgrade anyway"|trans({}, 'Admin.Actions') }}',
```
Just a question... Why "Upgrade anyway" is preceded by "**multiple**ModuleModalUpdateUpgrade" ? Because this wording is not linked with the number of modules but with the user being on maintenance mode. This wording could appear for the update of one **or** multiple modules if the user is already on maintenance mode.

I might be wrong but I just wanted to make sure to understand :)

NeOMakinG replied:
You're completely right, changing it

Commit:
Update upgrade anyway button key

Progi1984 reviewed (changes requested):
Don't forget to rebase : No "merge" commits in your PR.

SD1982 commented:
LGTM !!
**Maintenance mode enabled**
![image](https://user-images.githubusercontent.com/40635814/104324517-d7d26300-54e7-11eb-89ee-d555db62d05e.png)

**Maintenace mode disabled**
![image](https://user-images.githubusercontent.com/40635814/104324576-e7ea4280-54e7-11eb-83d9-d5aff9ef0495.png)

Thanks @NeOMakinG !!

Progi1984 commented:
Thanks @NeOMakinG & @SD1982
